### PR TITLE
Fixed issues with shortcut checking in 3d node editor

### LINKED ISSUE #54460: 3D editor does not recognize physical key inputs configured in freelook shortcuts (only non-physical keys work)
### Godot version

4.0.dev (1aef3a42b)

### System information

Fedora 34, GeForce GTX 1080 (NVIDIA 470.74)

### Issue description

The editor does not recognize physical key inputs configured in shortcuts (only non-physical keys).

This is now more important to fix as [inputs are now created as physical by default](https://github.com/godotengine/godot/pull/53707). The change also affects the editor shortcut dialog. We could disable the creation of physical key inputs in the editor shortcut dialog, but this would make it impossible to support freelook keys on non-QWERTY layouts by default ([something we should be able to fix now](https://github.com/godotengine/godot/issues/11777)).

**Edit:** After testing for https://github.com/godotengine/godot/issues/54469, I noticed that this only applies to freelook shortcuts in the 3D editor. For instance, this doesn't occur with the **Always Show Grid** shortcut in the 2D editor.

This is the part of the code that handles 3D editor freelook shortcuts: https://github.com/godotengine/godot/blob/1aef3a42b234b86c5db21d5a64509453e6bf145c/editor/plugins/node_3d_editor_plugin.cpp#L2606-L2623

cc @EricEzaM 

### Steps to reproduce

- Use a non-QWERTY keyboard layout such as AZERTY:
- Add a physical key shortcut to a 3D editor freelook action as follows:

image

- Try to press <kbd>Z</kbd> (on AZERTY) to move forward while having freelook enabled.

### Minimal reproduction project

N/A

### LINKED ISSUE #54469: 3D editor freelook only recognizes the first key assigned to a shortcut
### Godot version

4.0.dev (1aef3a42b)

### System information

Fedora 34, GeForce GTX 1080 (NVIDIA 470.74)

### Issue description

Related to https://github.com/godotengine/godot/issues/54460.

Following the merge of https://github.com/godotengine/godot/pull/51273, the editor only recognizes the first key assigned to a shortcut for freelook shortcuts (even if none of the keys are marked as physical).

Here, only <kbd>W</kbd> will be used to move forward while freelook is active:

image

Here, only <kbd>Up</kbd> will be used to move forward while freelook is active:

image

This does **not** occur with other shortcuts, such as the 2D editor's **Always Show Grid** shortcut:

image

Here, both <kbd>#</kbd> and <kbd>L</kbd> work as expected.

This is the part of the code that handles 3D editor freelook shortcuts: https://github.com/godotengine/godot/blob/1aef3a42b234b86c5db21d5a64509453e6bf145c/editor/plugins/node_3d_editor_plugin.cpp#L2606-L2623

### Steps to reproduce

Open the **Editor Settings**, go to the **Shortcuts** tab and add a second shortcut for one of the freelook movement keys.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by EricEzaM**
It works, but I would like feedback on the solution. If possible, please provide feedback. I think it is relatively clean and leverages existing systems rather that adding a bunch of code for this specific use case.

The freelook shortcuts now use the action system by proxy. This allows the actions system to be leveraged for input checking, as checking for shortcuts outside of `xxxx_input()` methods is not well supported. When the shortcut changes, the respective action is updated.

Please see my comments here for additional context: https://github.com/godotengine/godot/issues/54469#issuecomment-957693323

Fixes #54469 (all shortcuts are recognised)
Fixes #54460 (physical keys work now)

P.S. I will add additional in-code comments documenting _why_ this is done like this, if this solution is generally agreed with.

### CODE REVIEW

**Review by JFonS — APPROVED** [review]
Looks good from my side, but it's mostly input related changes, which I'm not very familiar with.

File: editor/plugins/node_3d_editor_plugin.cpp [hunk 1]
```diff
@@ -4561,6 +4567,15 @@ Node3DEditorViewport::Node3DEditorViewport(Node3DEditor *p_spatial_editor, Edito
...
+	register_shortcut_action("spatial_editor/freelook_forward", TTR("Freelook Forward"), Key::W);
+	register_shortcut_action("spatial_editor/freelook_backwards", TTR("Freelook Backwards"), Key::S);
+	register_shortcut_action("spatial_editor/freelook_up", TTR("Freelook Up"), Key::E);
+	register_shortcut_action("spatial_editor/freelook_down", TTR("Freelook Down"), Key::Q);
+	register_shortcut_action("spatial_editor/freelook_speed_modifier", TTR("Freelook Speed Modifier"), Key::SHIFT);
+	register_shortcut_action("spatial_editor/freelook_slow_modifier", TTR("Freelook Slow Modifier"), Key::ALT);
+
 	ED_SHORTCUT("spatial_editor/freelook_left", TTR("Freelook Left"), Key::A);
 	ED_SHORTCUT("spatial_editor/freelook_right", TTR("Freelook Right"), Key::D);
 	ED_SHORTCUT("spatial_editor/freelook_forward", TTR("Freelook Forward"), Key::W);
```
Comment (akien-mga) [inline editor/plugins/node_3d_editor_plugin.cpp:4581]: Aren't those redundant now? `register_shortcut_action` seems to do that + more.

Comment (EricEzaM) [reply editor/plugins/node_3d_editor_plugin.cpp:4581]: Hmm yep, looks like that could have been a re basing mistake.

File: editor/plugins/node_3d_editor_plugin.cpp [hunk 2]
```diff
@@ -4427,6 +4411,29 @@ void Node3DEditorViewport::finish_transform() {
 	surface->update();
 }
 
+
+// Register a shortcut and also add it as an input action with the same events.
```
[the remark this answers is not in the document]
Comment (akien-mga) [reply editor/plugins/node_3d_editor_plugin.cpp]: Make sure that you have clang-format 13 - but if you had an earlier version you'd get a lot of error spam due to a config we set which is only available in 13, so it sounds more like your pre-commit hook might not be running.

**Pull request merged**